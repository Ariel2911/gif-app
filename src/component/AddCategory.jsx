import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputValue = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      onNewCategory(inputValue);
      setInputValue('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        value={inputValue}
        onChange={(e) => handleInputValue(e)}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
