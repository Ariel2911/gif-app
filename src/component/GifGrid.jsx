import { GifGridItem } from './GridGifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>

      {isLoading && <p>Cargando...</p>}

      <div className="card-grid">
        {images.map(({ id, title, url }) => (
          <GifGridItem key={id} title={title} url={url} />
        ))}
      </div>
    </>
  );
};
