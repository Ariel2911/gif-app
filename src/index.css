* {
  font-family: Helvetica, Arial, sans-serif;
  background-color: rgb(234, 234, 234);
  box-sizing: border-box;
}

input {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgb(97, 32, 158);
  color: black;
  font-size: 1.2rem;
  outline: none;
  padding: 10px 15px;
  width: 100%;
}

h2 {
  font-size: 1.5rem;
}

h3 {
  font-size: 3rem;
  margin-bottom: 5px;
}

.card-grid {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.card {
  display: flex;
  flex-direction: column;
  align-self: center;
  width: 160px;
  margin: 8px;
  border-radius: 8px;
  background-color: white;
  box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
  overflow: hidden;
}

.card p {
  background-color: white;
  flex: 1;
  margin-top: 5px;
  padding: 5px 20px 0px 20px;
  text-align: center;
}

.card img {
  width: 100%;
}
