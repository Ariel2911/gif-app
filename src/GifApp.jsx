import { useState } from 'react';
import { AddCategory, GifGrid } from './component';

export const GifApp = () => {
  const [categories, setCategories] = useState(['saitama']);

  const onAddCategory = (newCategory) => {
    !categories.includes(newCategory) &&
      setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1>GifApp</h1>

      <AddCategory onNewCategory={onAddCategory} />

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
