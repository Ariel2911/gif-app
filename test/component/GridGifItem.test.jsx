import { render } from '@testing-library/react';
import { GifGridItem } from '../../src/component';

describe('Pruebas en <GridGifItem />', () => {
  const title = 'Saitama';
  const url = 'https://';

  test('debe de hacer match con el snapshop', () => {
    const { container } = render(<GifGridItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });
});
