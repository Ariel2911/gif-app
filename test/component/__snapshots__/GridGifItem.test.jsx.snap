// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GridGifItem /> debe de hacer match con el snapshop 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
